pkg/collector/process.go

```diff
@@ -130,41 +129,50 @@ func (cp *collectingProcess) decodePacket(packetBuffer *bytes.Buffer) (*entities
 func (cp *collectingProcess) decodeTemplateSet(templateBuffer *bytes.Buffer, obsDomainID uint32) (interface{}, error) {
 	var templateID uint16
 	var fieldCount uint16
-	err := decode(templateBuffer, &templateID, &fieldCount)
+	err := util.Decode(templateBuffer, &templateID, &fieldCount)
 	if err != nil {
 		return nil, fmt.Errorf("Error in decoding message: %v", err)
 	}
-	elements := make([]*templateField, 0)
-	templateMsg := entities.NewTemplateSet()
+	elements := make([]entities.InfoElement, 0)
+	templateSet := entities.NewTemplateSet()
 
 	for i := 0; i < int(fieldCount); i++ {
-		element := templateField{}
+		var element entities.InfoElement
+		var enterpriseID uint32
+		var elementID uint16
 		// check whether enterprise ID is 0 or not
-		elementID := make([]byte, 2)
+		elementid := make([]byte, 2)
 		var elementLength uint16
-		err = decode(templateBuffer, &elementID, &elementLength)
+		err = util.Decode(templateBuffer, &elementid, &elementLength)
 		if err != nil {
 			return nil, fmt.Errorf("Error in decoding message: %v", err)
 		}
-		indicator := elementID[0] >> 7
-		if indicator != 1 {
-			element.elementID = binary.BigEndian.Uint16(elementID)
-			element.enterpriseID = uint32(0)
-			element.elementLength = elementLength
+		isIANARegistry := elementid[0] >> 7
+		if isIANARegistry != 1 {
+			elementID = binary.BigEndian.Uint16(elementid)
+			enterpriseID = uint32(0)
+			element, err = registry.GetInfoElementFromID(elementID, enterpriseID)
+			if err != nil {
+				return nil, err
+			}
 		} else {
-			err = decode(templateBuffer, &element.enterpriseID)
+			err = util.Decode(templateBuffer, &enterpriseID)
 			if err != nil {
 				return nil, fmt.Errorf("Error in decoding message: %v", err)
```

Comment:
The error is returned when decoding has problems. Do you mean that we should return the plain error without saying `Error in ...`?

Reply:
I meant why not `return nil, err` like the if block above.

Reply:
Since it is a decoding error, I didn't specify the error type in decoding function. Thanks for pointing this out. I will need to change the error log in decode function and just return `nil, err` here.